storage/innobase/xtrabackup/test/t/bug1551634.sh

```diff
@@ -0,0 +1,87 @@
+########################################################################
+# Bug #1551634: xtrabackup --slave-info does not handle multi-master replication
+########################################################################
+# Plan:
+#  Set up 2 masters and 1 slave.
+#  Insert some data on both masters
+#  Ensure that slave1 has the data
+#  Backup the slave1
+#  Verify that xtrabackup_slave_info contains info about 2 channels.
+
+. inc/common.sh
+
+require_server_version_higher_than 5.7.6
+
+function init_db()
+{
+  local db_name=$1
+  local table_name=$2
+
+mysql <<EOF
+CREATE DATABASE IF NOT EXISTS $db_name;
+USE $db_name;
+CREATE TABLE IF NOT EXISTS $table_name (
+  c int(11) NOT NULL PRIMARY KEY
+) ENGINE=InnoDB DEFAULT CHARSET=latin1;
+EOF
+}
+
+readonly master1_id=1
+readonly master2_id=2
+readonly slave_id=3
+
+function run_test()
+{
+    local mode=$1
+
+    vlog "############################################################"
+    vlog "$mode mode"
+
+    if [ $mode = "GTID" ]
+    then
+        local extra_arguments="--gtid_mode=ON --enforce-gtid-consistency=ON"
+    else
+        local extra_arguments=
+        mode=
+    fi
+
+    vlog "############################################################"
+    vlog "Setting up masters and slave"
+    start_server_with_id $master1_id $extra_arguments
+    init_db test m1
+    multi_row_insert test.m1 \({1..7}\)
+
+    start_server_with_id $master2_id $extra_arguments
+    init_db test m2
+    # number of rows differ from one for master1 to get different MASTER_LOG_POS in xtrabackup_slave_info
+    multi_row_insert test.m2 \({1..13}\)
+
+    start_server_with_id $slave_id \
+        --master-info-repository=TABLE \
```

Comment:
That is required to have multiple channels on slave. If we start slave server without both         `--master-info-repository=TABLE` and `--relay-log-info-repository=TABLE \`, then setting up server as slave with `CHANGE MASTER ... FOR CHANNEL '...'` fails with:
`ERROR 3077 (HY000) at line 1: To have multiple channels, repository cannot be of type FILE; Please check the repository configuration and convert them to TABLE.`